doc/connection_and_messaging_reliability.md

```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
+```
 
 On [iothub\_client\_ll](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L402) module:
 
+```c
 IOTHUB\_CLIENT\_RESULT IoTHubClient\_LL\_SetConnectionStatusCallback(IOTHUB\_CLIENT\_LL\_HANDLE iotHubClientHandle, IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK connectionStatusCallback, void\* userContextCallback);
+```
 
 This callback will be invoked in these specific situations:
 
 - When the device client gets connected or re-connected to the Azure IoT Hub;
 
 - When the device client gets disconnected due to any issues;
 
-> These include network availability and IoT hub connectivity issues, authentication failures.
+  > These include network availability and IoT hub connectivity issues, authentication failures.
 
 - When the device client ceases attempting to re-connect (if the Retry Policy no longer allows to).
 
-Please take a look [here](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L144) for more info on the possible status and reasons.
+The type `IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK` is [defined](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L183) as:
+```c
+typedef void(*IOTHUB_CLIENT_CONNECTION_STATUS_CALLBACK)(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* userContextCallback);
+```
+
+The user application must provide a function that matches the function pointer definition above to provide it as an argument to any of the `*_SetConnectionStatusCallback` functions above.
+
+For reference, the connection status and reason enumerations (`IOTHUB_CLIENT_CONNECTION_STATUS` and `IOTHUB_CLIENT_CONNECTION_STATUS_REASON`, respectively) are defined in [iothub_client_core_common.h](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/inc/iothub_client_core_common.h#L143).
+
+
+#### Understanding IOTHUB_CLIENT_CONNECTION_STATUS
+
+Indicates whether the device client is connected or not to the Azure IoT Hub.
+
+|Value|Description|
+|-|-|
+|IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED|Effectively means the device client is not ready to communicate with the Azure IoT Hub. The device client could be in any state from completely disconnected to not yet authenticated, including when brief disconnections occur for SAS token refreshes. See the list of IOTHUB_CLIENT_CONNECTION_STATUS_REASON status below for further details.|
+|IOTHUB_CLIENT_CONNECTION_AUTHENTICATED|The device client is **ready** to communicate with the Azure IoT Hub, being both connected and authenticated.|
+
+#### Understanding IOTHUB_CLIENT_CONNECTION_STATUS_REASON
+
+This enumeration attemps to provide a more specific reason for the current connection status of the device client. Its values depend on the transport protocol chosen by the user application for the Azure IoT C SDK client (AMQP, MQTT or HTTP) and on error granularity provided by the Azure IoT Hub.
```

Comment:
"attempts to provide" - I think we can just say "provide".

Reply:
Well, it doesn't provide completely... I think it's more honest to say `attempts`, but I will simplify and change as you suggested.

Reply:
Fixed.